crypto/bn/bn_lib.c

```diff
@@ -437,7 +437,7 @@ static BIGNUM *bin2bn(const unsigned char *s, int len, BIGNUM *ret,
                       endianess_t endianess, signedness_t signedness)
 {
     int inc;
-    const unsigned char *s2;
+    const unsigned char *s2 = NULL;
```

Comment:
This really does not make any sense as:
1. the endianess enum has only two values BIG and LITTLE, both of them are covered by the `switch()` below.
2. if the endianess value was anything else the NULL value in s2 would cause segfault anyway on the dereference of s2 after the switch.
3. inc, inc2 is kept uninitialized 

The only (slightly reasonable) way would be to choose one of the endianess values as default, initialize the variables according to it and then transform the switch below to `if (endianess == the other value)`

Reply:
I think VS is just not smart enough to recognize this is a static function and all call sites within the c source is going to call bin2bn() with either BIG or LITTLE. So in theory it is impossible to dereference the null pointer. This is kind of a false positive but we need to satisfy the compilter unfortunately. However, given endianess is binary, perhaps a if statement is better than a switch statement.